drivers: can: stm32_bxcan: fix filter config

ttwards opened the pull request:
drivers: can: stm32_bxcan: fix filter config
Setting the filter registers in master CAN requires initializing master
CAN first. CONFIG_CAN_MAX_EXT_ID_FILTER banks are reserved for IDE
frames. Previously we set FS1R(CAN filter scale register) at init time,
but it is possible that the master CAN is not initialized at that time.
That is when the filter banks are not set correctly, causing that we get
wrong filter_id from slave CAN. This patch fixes the issue by setting
FS1R at the time of initializing master CAN.

Signed-off-by: Wenxi Xu <xuwenxi0517@gmail.com>

ttwards commented:
Hello there, is it possible that you take a look here?

ttwards commented:
> > Hello there, is it possible that you take a look here?
> 
> If you need specific people to review, you should @ mention them, otherwise I recommend joining discord and pinging the pr-help chat at https://discord.com/channels/720317445772017664/997527108844798012

Thank you, I’ll take a look there. Is it normal that it takes so long for a PR to merge?

henrikbrixandersen commented:
> Thank you, I’ll take a look there. Is it normal that it takes so long for a PR to merge?

You opened it just 8 days ago, just as we were releasing Zephyr v4.1.0. I will get to it.

ttwards commented:
> > Thank you, I’ll take a look there. Is it normal that it takes so long for a PR to merge?
> 
> You opened it just 8 days ago, just as we were releasing Zephyr v4.1.0. I will get to it.

That's very kind of you, thanks.

ttwards commented:
> > Thank you, I’ll take a look there. Is it normal that it takes so long for a PR to merge?
> 
> You opened it just 8 days ago, just as we were releasing Zephyr v4.1.0. I will get to it.

Hi there, could you please do some quick review?

ttwards commented:
Could you clarify which test configurations are missing? (e.g., specific hardware setups or edge cases?) I'll add them promptly.

henrikbrixandersen commented:
> Could you clarify which test configurations are missing? (e.g., specific hardware setups or edge cases?) I'll add them promptly.

Testing at my side on various STM32 boards with different CAN configurations.

henrikbrixandersen reviewed (changes requested):
You need to remove the `bank_offset` variable as it is now unused.

etienne-lms commented on drivers/can/can_stm32_bxcan.c:
```diff
@@ -645,10 +645,17 @@ static int can_stm32_init(const struct device *dev)
+					 << CAN_STM32_NUM_FILTER_BANKS;
+#endif /* DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) > 1 */
+
 	cfg->master_can->FMR &= ~CAN_FMR_FINIT;
```
By the way, fix indentation here

ttwards commented:
I guess it's problem with Github action? Or is there anything that I have to edit after v4.1.0 update?
```
# From Github Action
Run pip install -r scripts/requirements-actions.txt --require-hashes
  pip install -r scripts/requirements-actions.txt --require-hashes
  shell: /usr/bin/bash -e {0}
  env:
    pythonLocation: /opt/hostedtoolcache/Python/3.12.9/x64
    PKG_CONFIG_PATH: /opt/hostedtoolcache/Python/3.12.9/x64/lib/pkgconfig
    Python_ROOT_DIR: /opt/hostedtoolcache/Python/3.12.9/x64
    Python2_ROOT_DIR: /opt/hostedtoolcache/Python/3.12.9/x64
    Python3_ROOT_DIR: /opt/hostedtoolcache/Python/3.12.9/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.12.9/x64/lib
ERROR: Could not open requirements file: [Errno 2] No such file or directory: 'scripts/requirements-actions.txt'
Error: Process completed with exit code 1.
```

henrikbrixandersen commented:
Please rebase on `main`.

etienne-lms commented on drivers/can/can_stm32_bxcan.c:
```diff
@@ -645,10 +644,19 @@ static int can_stm32_init(const struct device *dev)
+		cfg->master_can->FS1R |= ((1U << CONFIG_CAN_MAX_EXT_ID_FILTER) - 1);
+
+#if DT_NUM_INST_STATUS_OKAY(DT_DRV_COMPAT) > 1
+		/* reserve ext_id filters on slave can device */
```
nitpicking: s/can/CAN/  (sorry I missed it in my previous review)

ttwards replied:
Thanks for catching this! Fixed in the latest commit (s/can/CAN/ in the comment).

ttwards committed:
drivers: can: stm32_bxcan: fix filter config

Setting the filter registers in master CAN requires initializing master
CAN first. CONFIG_CAN_MAX_EXT_ID_FILTER banks are reserved for IDE
frames. Previously we set FS1R(CAN filter scale register) at init time,
but it is possible that the master CAN is not initialized at that time.
That is when the filter banks are not set correctly, causing that we get
wrong filter_id from slave CAN. This patch fixes the issue by setting
FS1R at the time of initializing master CAN.

Tested on:
STM32F407IGH6 with 2 std_id and 2 ext_id on each of CAN1 and CAN2.

Signed-off-by: Wenxi Xu <xuwenxi0517@gmail.com>
